agent/handlers/agentapi/taskprotection/v1/handlers/handlers.go

```diff
@@ -222,12 +221,51 @@ func GetTaskProtectionHandler(state dockerstate.TaskEngineState,
 			return
 		}
 
-		// TODO: Call ECS
 		logger.Info("GetTaskProtection endpoint was called", logger.Fields{
 			loggerfield.Cluster: cluster,
 			loggerfield.TaskARN: task.Arn,
 		})
-		writeJSONResponse(w, http.StatusOK, "Ok", getTaskProtectionRequestType)
+
+		ecsClient, responseCode, err := factory.newTaskProtectionClient(credentialsManager, task)
+		if err != nil {
+			writeJSONResponse(w, responseCode, err.Error(), getTaskProtectionRequestType)
+			return
+		}
+
+		response, err := ecsClient.GetTaskProtection(&ecs.GetTaskProtectionInput{
+			Cluster: aws.String(cluster),
+			Tasks:   aws.StringSlice([]string{task.Arn}),
+		})
+
+		if err != nil {
+			exceptionType, statusCode := getExceptionTypeAndStatusCode(err)
+			logger.Error(fmt.Sprintf("ECS throws an exception with type %s", exceptionType))
```

Comment:
How about we put `err` in the log?

Reply:
With log.Error we should be able to see an [ERROR] before the log line. I think it should be sufficient for searching purpose.

Reply:
I mean the `err` value.

Reply:
Oh yeah I can (and should) add that.